### DISCUSSION TITLE
is there any way to apply Event Subscriber in Seeder ?

### DISCUSSION THREAD

**Discussion by superiums** [mikro-orm/mikro-orm#4631] [General]
the event subsciber usually holds some logic to keep the data to be clean and completely.
eg. we can auto fill the path field of an pivot table.

so, i wonder if there's a way to apply Event Subscriber in Seeder.

**Comment by B4nan** [comment]
It should just work, do you have an actual problem with it?

**Reply by superiums** [reply]
as mentioned in the doc, we could apply the subscriber like this : 
```ts
MikroORM.init({
  subscribers: [new AuthorSubscriber()],
});
```

but i launch the seeder by CLI, like this `pnpm mikro-orm seeder:run  `

so how could i register the subscriber?

**Reply by B4nan** [reply]
You need to have a CLI config and put it there.

**Reply by superiums** [reply]
is there any doc or example?
seems there's no config avaluable like : `subscribers` in mikro-orm.config.ts

**Reply by B4nan** [reply]
The `MikroORM.init` method accepts the exact same thing, not sure why you think its not there.

**Reply by superiums** [reply]
i'm following doc in : [https://mikro-orm.io/docs/next/seeding](https://mikro-orm.io/docs/next/seeding)

- i have a seeder like this:
./database/seeder/database.seeder.ts

```ts
./database/seeder/database.seeder.ts
import { EntityManager } from '@mikro-orm/core';
import { Seeder } from '@mikro-orm/seeder';
import { Author } from './author'

export class DatabaseSeeder extends Seeder {

  async run(em: EntityManager): Promise<void> {
    // will get persisted automatically
    const author = em.create(Author, {
      name: 'John Snow',
      email: 'snow@wall.st'
    });

    // but if we would do `const author = new Author()` instead,
    // we would need to call `em.persist(author)` explicitly.
  }
}
```

- i'm running it via `npx mikro-orm seeder:create DatabaseSeeder`

so i don't know where to found the mikroorm.init . seems it was hanldled by mikro automatically.

**Reply by B4nan** [reply]
You need to put the subscribers to the config, since you use CLI, it will use the CLI config.

https://mikro-orm.io/docs/installation#setting-up-the-commandline-tool
